device-server/src/main/kotlin/com/badoo/automation/deviceserver/host/HostFactory.kt

```diff
@@ -21,41 +23,44 @@ class HostFactory(
     private val logger = LoggerFactory.getLogger(javaClass.simpleName)
 
     override fun getHostFromConfig(config: NodeConfig): ISimulatorsNode {
-        logger.info("Trying to start node $config.")
-
-        val hostName = config.host
-        val userName = config.user
-        val publicHostName = config.publicHost
-        val remote: IRemote = remoteProvider(hostName, userName, publicHostName)
-
-        if (userName.isBlank() && !remote.isLocalhost()) {
-            throw RuntimeException("Config for non-localhost nodes must have non-empty 'user'. Current config: $config")
+        logger.info("Obtaining node from $config.")
+        try {
+            return when {
+                config.type == NodeConfig.NodeType.Emulators -> EmulatorsNode(
```

Comment:
Maybe make this ```NodeConfig.NodeType.AndroidEmulators``` (and ```NodeConfig.NodeType.IosSimulators``` as well) ?

Reply:
I'll make it AndroidEmulators; I'm not going to change iOS stuff - that's a different task.